x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -257,6 +277,17 @@ protected LogicalPlan rule(Aggregate aggregate) {
             return plan;
         }
 
+        private static Block SINGLE_NULL_VALUE = BlockUtils.constantBlock(PlannerUtils.NON_BREAKING_BLOCK_FACTORY, null, 1);
+
+        /**
+         * Create an (essentially) empty row; the row still has 1 placeholder attribute since plans with empty schemas are pruned.
+         */
+        private static LocalRelation emptyRow(Source source) {
```

Comment:
This isn't necessary as the block is always the same; I'm pretty sure we have something similar declared somewhere else inside the code.
Just create the relationship in place no need for an emptyRow method.
Speaking of, propagate the source instead of passing in `Source.EMPTY` - that's never a good solution inside the planner.

Reply:
Inlined and shortened!